src/install/fetchResolution.ts

```diff
@@ -6,6 +6,9 @@ import execa = require('execa')
 const gitLogger = logger('git')
 
 export default async function fetchRes (res: ResolveResult, target: string, opts: FetchOptions): Promise<void> {
+  if (res.fetch) {
+      return res.fetch(target)
+  }
```

Comment:
This is so custom resolvers can override fetch and still specify `tarball` and `repo`.